src/main/java/org/fcrepo/importexport/common/TransferProcess.java

```diff
@@ -81,49 +81,75 @@ public static String decodePath(final String encoded) {
      * @param baseDir the base directory in the export package
      * @return a unique location for binary resources from the path of the given URI
      *         would be stored
      */
-    public static File fileForBinary(final URI uri, final File baseDir) {
-        return fileForURI(uri, baseDir, BINARY_EXTENSION);
+    public static File fileForBinary(final URI uri, final String sourcePath, final String destinationPath,
+            final File baseDir) {
+        return fileForURI(uri, sourcePath, destinationPath, baseDir, BINARY_EXTENSION);
     }
 
     /**
      * Gets the file where an external binary resource at the given URL would be stored
      * in the export package.
      * @param uri the URI for the resource
+     * @param sourcePath the base path the resource was exported from
+     * @param destinationPath the base path the resource is importing to
      * @param baseDir the base directory in the export package
      * @return a unique location for external resources from the path of the given URI
      *         would be stored
      */
-    public static File fileForExternalResources(final URI uri, final File baseDir) {
-        return fileForURI(uri, baseDir, EXTERNAL_RESOURCE_EXTENSION);
+    public static File fileForExternalResources(final URI uri, final String sourcePath,
+            final String destinationPath, final File baseDir) {
+        return fileForURI(uri, sourcePath, destinationPath, baseDir, EXTERNAL_RESOURCE_EXTENSION);
     }
 
     /**
      * Gets the file where a resource at the given URL with the given extension
      * would be stored in the export package.
      * @param uri the URI for the resource
+     * @param sourcePath the base path the resource was exported from
+     * @param destinationPath the base path the resource is importing to
      * @param baseDir the baseDir directory in the export package
      * @param extension the arbitrary extension expected the file
      * @return a unique location for resources from the path of the given URI
      *         would be stored
      */
-    public static File fileForURI(final URI uri, final File baseDir, final String extension) {
-        return new File(baseDir, TransferProcess.encodePath(uri.getPath()) + extension);
+    public static File fileForURI(final URI uri, final String sourcePath, final String destinationPath,
+            final File baseDir, final String extension) {
+        return new File(baseDir, encodePath(relativePath(uri, sourcePath, destinationPath)) + extension);
+    }
+
+    /**
+     * Translates the path of a URI relative to an optional source URI.
+     * @param uri the URI for the resource
+     * @param sourcePath the base path the resource was exported from
+     * @param destinationPath the base path the resource is importing to
+     */
+    public static String relativePath(final URI uri, final String sourcePath, final String destinationPath) {
```

Comment:
The `TransferProcess` interface is used by both the `Importer` and `Exporter`... however, this method appears to be written in the context of the `Exporter`... whereas the analogous `Importer` capability is found here:
https://github.com/fcrepo4-labs/fcrepo-import-export/pull/81/files#diff-f6498b3791251ba98aa82e1e70ebbe51R512

Would it be more clear to name this method differently?